main.go

```diff
@@ -110,6 +124,42 @@ func main() {
 		log.Printf("Overlapping routes %v with the range %s", routes, natV4Range)
 	}
 
+	if hostname == "" {
+		hostname, err = os.Hostname()
+		if err != nil {
+			log.Fatalf("Cannot fetch os.Hostname: %v", err)
+		}
+		log.Printf("No hostname specified, using os.Hostname: %s", hostname)
+	}
+
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		log.Fatalf("Cannot fetch cluster config: %v", err)
+	}
+	k8sClient, err := kubernetes.NewForConfig(config)
+
+	if len(podCIDR) == 0 {
+		log.Printf("Watching for node, awaiting podCIDR allocation")
+		ctx := context.Background()
+		node, err := waitForPodCIDR(ctx, k8sClient, hostname)
+		if err != nil {
+			log.Fatalf("waitForPodCIDR: %v", err)
+		}
+		// take podCIDR from primary IP family configured for the cluster
+		podCIDR = node.Spec.PodCIDRs[0]
+		log.Printf("podCIDR: %s", podCIDR)
+	}
+
+	_, _, podMaskSize, err := parsePodCIDR(podCIDR)
+	if err != nil {
+		log.Fatalf("Wrong pod CIDR %s: %v", podCIDR, err)
+	}
+	// TODO: it's fixed to /112 now because IPv4 range for source addresses for NAT is hardcoded
+	//       right now, this will be later changed to depend on the size of that range once it's configurable
+	if podMaskSize < widestPodCIDRRangeAllowed {
+		log.Fatalf("Mask for pod CIDR must be /120 or narrower to avoid source address collision for NAT64: %s", podCIDR)
+	}
+
```

Comment:
just parse the podCIDR here and operate with that, from this point 

```suggestion
        _, podIPNet, err := net.ParseCIDR(podCIDR)
	if err != nil {
		return nil, nil, 0, fmt.Errorf("net.ParseCIDR: %w", err)
	}
	if podIPNet.IP.To4() != nil {
		return nil, nil, 0, fmt.Errorf("podCIDR is required to be IPv6 CIDR: %s", podCIDR)
	}
	// TODO: it's fixed to /112 now because IPv4 range for source addresses for NAT is hardcoded
	//       right now, this will be later changed to depend on the size of that range once it's configurable
	if podIPNet.Mask.Size() < widestPodCIDRRangeAllowed {
		log.Fatalf("Mask for pod CIDR must be /120 or narrower to avoid source address collision for NAT64: %s", podCIDR)
	}

```